packages/evm/contracts/BoostCore.sol

```diff
@@ -204,6 +233,90 @@ contract BoostCore is Ownable, ReentrancyGuard {
         return _boosts.length;
     }
 
+    /// @notice Get the incentives for a Boost
+    /// @param key The key composed of the Boost ID and the Incentive ID - keccak256(abi.encodePacked(boostId, incentiveId))
+    function getIncentive(bytes32 key) external view returns (IncentiveDisbursalInfo memory) {
+        return incentives[key];
+    }
+
+    /// @notice Returns the protocol fee and any remaining incentive value to the owner or budget
+    /// @param boostId The ID of the Boost
+    function clawback(bytes calldata data_, uint256 boostId, uint256 incentiveId) external nonReentrant {
+        BoostLib.Boost memory boost = _boosts[boostId];
+
+        if(msg.sender != address(boost.budget)) {
+            revert BoostError.Unauthorized();
+        }
+
+        // Generate the unique key for the incentive
+        bytes32 key = _generateKey(boostId, incentiveId);
+        IncentiveDisbursalInfo storage incentive = incentives[key];
+
+        // Decode the data for clawback
+        AIncentive.ClawbackPayload memory claim_ = abi.decode(data_, (AIncentive.ClawbackPayload));
+        uint256 amount = abi.decode(claim_.data, (uint256));
+
+        // Calculate the protocol fee based on the clawback amount and the protocol fee percentage
+        uint256 protocolFeeAmount = (amount * incentive.protocolFee) / FEE_DENOMINATOR;
+
+        // Transfer the protocol fee to the protocol fee receiver
+        if (protocolFeeAmount > 0) {
+            if (incentive.assetType == ABudget.AssetType.ERC20 || incentive.assetType == ABudget.AssetType.ETH) {
+                incentive.asset.safeTransfer(protocolFeeReceiver, protocolFeeAmount);
+            } else if (incentive.assetType == ABudget.AssetType.ERC1155) {
+                // wake-disable-next-line reentrancy (false positive, function is nonReentrant)
+                IERC1155(incentive.asset).safeTransferFrom(
+                    address(this), protocolFeeReceiver, incentive.tokenId, protocolFeeAmount, ""
+                );
+            }
+            emit ProtocolFeesCollected(boostId, incentiveId, protocolFeeAmount, protocolFeeReceiver);
+        }
+
+        bool success = boost.incentives[incentiveId].clawback(abi.encode(claim_));
```

Comment:
There is still a risk here that someone calims funds from an incentive directly, which means we'd have to recover their funds for them, but with the support of the sdk we could chalk this up as user error.